src/promptflow-tools/promptflow/tools/open_source_llm.py

```diff
@@ -51,6 +52,329 @@ def wrapper(*args, **kwargs):
+
+        for c in connections:
+            if c.type == ConnectionType.CUSTOM and "model_family" in c.configs:
+                try:
+                    validate_model_family(c.configs["model_family"])
+                    result.append({
+                        "value": f"connection/{c.name}",
+                        "display_value": f"[Connection] {c.name}",
+                        #"hyperlink": f"https://ml.azure.com/endpoints/realtime/{e.endpoint_name}/detail?wsid=/subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/providers/Microsoft.MachineLearningServices/workspaces/{workspace_name}",
+                        "description": f"Custom Connection:  {c.name}",
+                    })
+
+                except:
+                    # silently ignore unsupported model family
+                    continue
+
+        return result
+    
+    def get_local_custom_connection_names(self) -> List[Dict[str, Union[str, int, float, list, Dict]]]:
+        
+        result = []
+        try:
+            from promptflow import PFClient
+        except:
+            print("Skipping Local PFClient.")
+            return result
+
+        pf = PFClient()
+        connections = pf.connections.list()
+        
+        for c in connections:
+            if c.type == ConnectionType.CUSTOM and "model_family" in c.configs:
+                try:
+                    validate_model_family(c.configs["model_family"])
+                    result.append({
+                        "value": f"localConnection/{c.name}",
+                        "display_value": f"[Local Connection] {c.name}",
+                        #"hyperlink": f"https://ml.azure.com/endpoints/realtime/{e.endpoint_name}/detail?wsid=/subscriptions/{subscription_id}/resourceGroups/{resource_group_name}/providers/Microsoft.MachineLearningServices/workspaces/{workspace_name}",
+                        "description": f"Local Custom Connection:  {c.name}",
+                    })
+
+                except:
+                    # silently ignore unsupported model family
+                    continue
+
+        return result
+    
+    def get_endpoint_from_local_custom_connection(self, connection_name) -> Tuple[str, str, str]:
+        from promptflow import PFClient
+        pf = PFClient()
+        
+        connection = pf.connections.get(connection_name, with_secrets=True)
+        
+        return self.get_endpoint_from_custom_connection(connection)
+   
+    def get_endpoint_from_azure_custom_connection(self,
+                            subscription_id,
+                            resource_group_name,
+                            workspace_name,
+                            connection_name) -> Tuple[str, str, str]:
+        from promptflow.azure import PFClient
+        from azure.identity import DefaultAzureCredential
+        credential = DefaultAzureCredential(exclude_interactive_browser_credential=False)
+
+        azure_pf_client = PFClient(
```

Comment:
@brynn-code to help review the way to get connection.

Reply:
@brynn-code , my tests are running fine locally using connections.json. However when they run on PR they fail with this error. Am I doing something wrong in how locall connections are being pulled? I think CI test setup does not recognize these as locall connections.
![image](https://github.com/microsoft/promptflow/assets/75808410/0f574bd6-d2de-4aa5-ab8a-360072a51a64)

Reply:
Will promptflow.azure be available in default runtime? Is this the right way to get cloud connections?

Reply:
Checked `azure-ai-ml` is installed in default runtime.

Reply:
Confirmed over call with, promptflow azure will be available in defauilt runtime and the usage of PFClient is also correct in the tool.